mcf-browser/src/back-end/parse-mcf.js

```diff
@@ -0,0 +1,258 @@
+    if (fileFullName.includes('.')) {
+      const fileNameSpiltOnPeriod = fileFullName.split('.');
+      fileNameSpiltOnPeriod.pop();
+      fileName = fileNameSpiltOnPeriod.join('.');
+    }
+    this.prov = 'local: ' + fileName;
+    this.curNode = null;
+    this.lineNum = -1;
+  }
+
+  /**
+   * Parses a string represetnaing a comma separated list of property values
+   * from a line of an mcf file. Returns a list having either a string or
+   * <namespace, reference> pair.
+   *
+   * @param {string} propValues A comma separated list of property values.
+   * @return {Array<(string|Object)>} Array of
+   *     parsed values.
+   */
+  parsePropValues(propValues) {
+    if (!propValues) {
+      throw new Error('Error, no property values to parse (line ' +
+                      this.lineNum + ')');
+    }
+    const values = [];
+    for (const propValue of propValues.split(',')) {
+      if (propValue.split(':').length === 2) {
+        const namespace = propValue.split(':')[0].trim();
+        if (NAMESPACES.indexOf(namespace) >= 0) {
+          values.push({
+            'ns' : namespace,
+            'ref' : propValue.split(':')[1].trim(),
+          });
+          continue;
+        }
+      }
+      values.push(propValue.trim());
+    }
+    return values;
+  }
+  /**
+   * Sets curNode variable of the calling ParseMcf object based on the passed in
+   * parsed values of a line of mcf, given the property label for the line was
+   * 'Node'. The parsed value either is a local reference with or without the
+   * 'l' namespace, or it has a remote namespace. If the namespace is remote,
+   * then the dcid for curNode is set. Updates localNodeHash mapping to store
+   * the subject nodes to be displayed in home screen of browser.
+   *
+   * @param {Array<string|Object>} parsedValues The array of parsed values from
+   *     a line of mcf with property label of 'Node'.
+   */
+  setCurNode(parsedValues) {
+    if (parsedValues.length !== 1) {
+      throw new Error('Error in declaring node (line ' + this.lineNum +
+                      '): ' + parsedValues.toString());
+    }
+
+    let isDCID = false;
+    let nodeId = parsedValues[0];
+
+    if (parsedValues[0] instanceof Object) {
+      nodeId = parsedValues[0]['ref'];
+      if (parsedValues[0]['ns'] !== LOCAL_NS) {
+        isDCID = true;
+      }
+    }
+
+    this.curNode = Node.getNode(nodeId, /* shouldCreate */ true);
+    if (isDCID) {
+      this.curNode.setDCID(parsedValues[0]['ref']);
+    }
+    ParseMcf.localNodeHash[nodeId] = this.curNode;
+  }
+
+  /**
+   * Sets the dcid of the curNode variable of the calling ParseMcf object given
+   * the property label of the line being parsed is 'dcid'.
+   * @param {Array<string|Object>} parsedValues The array of parsed values from
+   *     a line of mcf with property label of 'dcid'.
+   */
+  setCurNodeDCID(parsedValues) {
+    if (!this.curNode || parsedValues.length !== 1 ||
+        typeof parsedValues[0] !== 'string') {
```

Comment:
Have a TODO to handle each of these case separately and return meaningful errors?

(no repeated values vs. value must be string not ref, etc.)